doc/smart-switch/high-availability/smart-switch-ha-hld.md

```diff
@@ -0,0 +1,2176 @@
+
+##### 4.2.2.2. Tunneling packet to remote DPU
+
+<p align="center"><img alt="Tunneling packet to remote DPU" src="./images/npu-to-dpu-tunnel-remote.svg"></p>
+ 
+##### 4.2.2.3. Tunneling packet between DPUs
+
+This tunnel needs to be also implemented by DPU, because in planned or unplanned events, traffic might land on the in non-active side DPU and needs to be tunneled. This is using the same NPU-to-DPU traffic tunnel.
+ 
+<p align="center"><img alt="Tunneling packet between DPUs" src="./images/npu-to-dpu-tunnel-from-dpu.svg"></p>
+
+##### 4.2.2.4. Put all tunnels together
+
+Now let’s put all tunnels together. This can happen when the packets land on other switches that help traffic forwarding in the middle of HA state transition.
+ 
+<p align="center"><img alt="Tunneling packet between DPUs" src="./images/npu-to-dpu-tunnel-2-stages.svg"></p>
+
+#### 4.2.3. DPU-to-DPU communication channel
+
+There are 3 types of DPU-to-DPU communication channel: 
+
+- **Control Plane Control Channel**: This channel is used for transferring lightweighted data, such as control commands like HA state machine. This channel can be implemented by gRPC.
+- **Control Plane Data Channel**: This channel is used for doing heavy data transfer between DPUs, such as bulk sync and other communication that SONiC handles (syncd above SAI). This channel can also be done by gRPC.
+- **Data Plane Channel**: This channel is implemented below SAI, used for inline flow replication or any other network-triggered inline data sync.
+
+And to provide HA for DPU-to-DPU communication, we can also leverage our ECMP network setup.
+
+##### 4.2.3.1. HA Control Plane Channel
+
+###### 4.2.3.1.1. HA control plane control channel
+
+The control plane control channel is used for each DPU to talk to each other for things like state machine management. Implementation-wise, we will need a new service for HA communication and management, likely in its own container or SWSS.
+
+Having our own control channel is important. If we consider the switches as data plane and our upstream services as control plane, then data plane should be able to operate on its own, when control plane is disconnected. For example, when control plane is down and network is having trouble, we should still be able to react and adjust our services. Coupling data plane with control plane will introduce more failure modes also making the data plane unable to react properly.
+
+The DPU state machine management will run on NPU. Since each NPU manages multiple DPUs, all NPUs need to connect to all other NPUs, which forms a full mesh.
+
+The data path of this channel will look like below:
+<p align="center"><img alt="HA control plane control channel data path" src="./images/ha-control-plane-control-channel-data-path.svg"></p>
+
+Since DPU events are still coming from the DPU side via SAI/syncd, we will need to pass these events to the NPU side via ZMQ, which go through the PCIe bus.
```

Comment:
This is unclear to me. There are two levels of SONiC stacks, one on each DPU and one on NPU. So I don't know which SAI/syncd you're referring to here. I also don't know which "PCIe bus" you're describing, since presumably there could be PCIe between the CPU and NPU, and PCIe between CPU and each DPU.

Perhaps the spec needs a SW stack block diagram showing both of these in some appropriate level of detail. I'd like to understand the event flow you just described here including ZMQ role.

Reply:
Yep! I have updated the HA control plane overview section (see section 5.2 in the latest doc), which have this comment addressed. ZMQ is just a underlying communication channel we might or might not be using, which is not too important, so I have it removed.